app/src/main/java/com/capstone/weatherapp/CityListFragment.kt

```diff
@@ -30,9 +32,10 @@ class CityListFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        viewModel.getCities()
         rvCities = binding.rvCities
-        viewModel = CityListViewModel()
-        rvCities.adapter = CityListAdapter(viewModel.list)
+        cityListAdapter = CityListAdapter(viewModel.cityList)
+        rvCities.adapter = cityListAdapter
         rvCities.layoutManager = LinearLayoutManager(context)
     }
```

Comment:
viewModel.cityList.observe(viewLifecycleOwner) {
   adapter.updateData(it)
}
viewModel.getCities()